src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -681,8 +1215,377 @@ def infer(self, img, upsample_ratio=4, stride=8, track=True, smooth=True, multis
             else:
                 heatmap = self.prev_heatmap
         self.counter += 1
+        bounds = ([self.xmin, self.xmax, self.ymin, self.ymax],)
+        return current_poses, heatmap, bounds
+
+    def infer_adaptive(self, img, upsample_ratio=4, stride=8, track=True, smooth=True, multiscale=False):
+        """
+            This method is used to perform pose estimation on an image.
+
+            :param img: image to run inference on
+            :rtype img: engine.data.Image class object
+            :param upsample_ratio: Defines the amount of upsampling to be performed on the heatmaps and PAFs
+                when resizing,defaults to 4
+            :type upsample_ratio: int, optional
+            :param stride: Defines the stride value for creating a padded image
+            :type stride: int,optional
+            :param track: If True, infer propagates poses ids from previous frame results to track poses,
+                defaults to 'True'
+            :type track: bool, optional
+            :param smooth: If True, smoothing is performed on pose keypoints between frames, defaults to 'True'
+            :type smooth: bool, optional
+            :param multiscale: Specifies whether evaluation will run in the predefined multiple scales setup or not.
+            :type multiscale: bool,optional
+
+            :return: Returns a list of engine.target.Pose objects, where each holds a pose
+            and a heatmap that contains human silhouettes of the input image.
+            If no detections were made returns an empty list for poses and a black frame for heatmap.
+
+            :rtype: poses -> list of engine.target.Pose objects
+                    heatmap -> np.array()
+        """
```

Comment:
Indentation fix and removed unused arguments' descriptions.

```suggestion
        """
        This method is used to perform pose estimation on an image.

        :param img: image to run inference on
        :rtype img: engine.data.Image class object
        :param upsample_ratio: Defines the amount of upsampling to be performed on the heatmaps and PAFs
            when resizing,defaults to 4
        :type upsample_ratio: int, optional
        :param stride: Defines the stride value for creating a padded image
        :type stride: int,optional

        :return: Returns a list of engine.target.Pose objects, where each holds a pose
        and a heatmap that contains human silhouettes of the input image.
        If no detections were made returns an empty list for poses and a black frame for heatmap.

        :rtype: poses -> list of engine.target.Pose objects
                heatmap -> np.array()
        """
```